src/MizarData/include/MizarData/MizarPairedData.h

```diff
@@ -141,6 +142,40 @@ class MizarPairedDataTmpl {
   }
 
  private:
+  struct Frame {
+    TimestampNs start;
+    TimestampNs end;
+  };
+
+  // Op is a `void` callable that takes `Accumulator&` and `Frame`.
+  template <typename Accumulator, typename Op>
+  [[nodiscard]] Accumulator ReduceOverFrames(FrameTrackId frame_track_id,
+                                             RelativeTimeNs min_relative_time,
+                                             RelativeTimeNs max_relative_time, Op&& op) const {
+    Accumulator accumulator;
+    const auto [min_timestamp_ns, max_timestamp_ns] =
+        RelativeToAbsoluteTimestampRange(min_relative_time, max_relative_time);
+    const std::vector<TimestampNs> frame_starts =
+        GetFrameStarts(frame_track_id, min_timestamp_ns, max_timestamp_ns);
+    if (frame_starts.size() < 2) return accumulator;
+
+    for (size_t i = 0; i + 1 < frame_starts.size(); ++i) {
+      std::invoke(op, accumulator, Frame{frame_starts[i], frame_starts[i + 1]});
+    }
+    return accumulator;
+  }
+
+  [[nodiscard]] RelativeTimeNs FrameActiveInvocationTime(const absl::flat_hash_set<TID>& tids,
```

Comment:
TBH: I find this function super hard to read.
So you have a set of all thread-ids and a single frame (with a start and end timestamp).
I would hightly argue for a normal loop. Also, the `Times` function is quite inconvenient to read. Why can't we do sampling_period * callstack_count?

Reply:
> TBH: I find this function super hard to read. ... I would hightly argue for a normal loop. 

This reminds me of your earlier [advice](https://github.com/google/orbit/pull/4021#discussion_r931894950). There we needed a whole new structure to make `c_accumulate` readable -- and I agreed with you. Here the call to `transform_reduce` is as simple as it gets (barred the case of a named lambda). You're effectively arguing against using `transform_reduce`... anywhere. Has the committee made a mistake adding it? 

> Also, the Times function is quite inconvenient to read. Why can't we do sampling_period * callstack_count?

First of all, it's orthogonal to the PR. Overloaded operators only look simple. Typedefs allow for arbitrary arithmetics customisation. Hence, it was ruled operators should be replaced with free functions.

Reply:
... regarding operators also see TotW#84

> We mostly stick to the basics: assignment ("="), streaming ("<<"), call ("()"), and the relationals ("==", "!=", "<", ">", "<=", ">=")

Reply:
... regarding `reduce_transform` call

We can make use of implicit capture to improve readability. The style-guide recommends explicit captures only for the lambdas that might escape the scope.

Reply:
>First of all, it's orthogonal to the PR. Overloaded operators only look simple. Typedefs allow for arbitrary arithmetics customisation. Hence, it was ruled operators should be replaced with free functions.

I agree. Just mentioned it, because it still hindered readability. I guess this is more of an argument against strong typedefs. It should not part of this discussion here.

Reply:
>This reminds me of your earlier https://github.com/google/orbit/pull/4021#discussion_r931894950. There we needed a whole new structure to make c_accumulate readable -- and I agreed with you. Here the call to transform_reduce is as simple as it gets (barred the case of a named lambda). You're effectively arguing against using transform_reduce... anywhere. Has the committee made a mistake adding it?

I also remembered this discussion back then, and I can't tell you why in particular I found this "simple" code here hard to read. And I see that boils down to me arguing agains transform_reduce in general. Maybe it is fine and we can completely stick to the current version.

I also don't think it is the explicit capture that hinders readability.